Sources/StreamChat/Controllers/MemberListController/MemberListController.swift

```diff
@@ -140,13 +142,37 @@ public extension ChatChannelMemberListController {
         completion: ((Error?) -> Void)? = nil
     ) {
         var updatedQuery = query
+        let pagination = Pagination(pageSize: limit, offset: members.count)
         updatedQuery.pagination = Pagination(pageSize: limit, offset: members.count)
-        memberListUpdater.load(updatedQuery) { error in
-            self.query = updatedQuery
-            self.callback {
-                completion?(error)
+
+        membersLoader.executeInQueue(
+            executor: { [weak self] executorCompletion in
+                guard let self = self else {
+                    log.error("Callback called when self is nil")
+                    executorCompletion(nil)
+                    return
+                }
+
+                // Requests are put to serial queue,
+                // So check maybe we already requested and loaded this data before
+                if self.members.count > pagination.offset + pagination.pageSize {
+                    self.query = updatedQuery
+                    executorCompletion(nil)
+                    return
+                }
+                
+                // Perform Loading
+                self.memberListUpdater.load(updatedQuery) { [weak self] error in
+                    self?.query = updatedQuery
+                    executorCompletion(error)
+                }
```

Comment:
You will need the payload response of the `memberListUpdater` to check if there are more members or not. If the fetched payload.members.count is below the provided limit, it means there are no more members to fetch. Like mentioned above, please see the `ChannelController` implementation.